inference-engine/src/transformations/src/transformations/common_optimizations/nearest_neighbor_umpsampling_fusion.cpp

```diff
@@ -0,0 +1,322 @@
+    auto reshape_2 = pattern::wrap_type<opset8::Reshape>({mul, concat_2});
+
+    ngraph::matcher_pass_callback callback = [=](ngraph::pattern::Matcher& m) {
+        auto &pattern_to_output = m.get_pattern_value_map();
+
+        const auto reshape_2_node = std::dynamic_pointer_cast<opset8::Reshape>(pattern_to_output.at(reshape_2).get_node_shared_ptr());
+        const auto mul_node = std::dynamic_pointer_cast<opset8::Multiply>(pattern_to_output.at(mul).get_node_shared_ptr());
+
+        if (!reshape_2_node || !mul_node || mul_node->get_input_partial_shape(1).is_dynamic()) return false;
+
+        const auto mul_const_node = std::dynamic_pointer_cast<opset8::Constant>(mul_node->input_value(1).get_node_shared_ptr());
+        if (!mul_const_node) return false;
+
+        const auto reshape_1_node = std::dynamic_pointer_cast<opset8::Reshape>(pattern_to_output.at(reshape_1).get_node_shared_ptr());
+        if (!reshape_1_node || reshape_1_node->get_input_partial_shape(0).is_dynamic()) return false;
+
+        uint64_t input_rank = static_cast<uint64_t>(reshape_1_node->get_input_partial_shape(0).rank().get_length());
+        const auto mul_const_shape = mul_const_node->get_output_shape(0);
+        const auto scales = get_scales_from_mul_const_shape(mul_const_shape, input_rank);
+        if (scales.empty() || std::all_of(scales.begin(), scales.end(), [](float s) { return s == 1.0f;})) { return false; }
+
+        const auto mul_const_value = mul_const_node->cast_vector<float>();
+        if (std::any_of(mul_const_value.begin(), mul_const_value.end(), [](float x){ return x != 1.0f; })) { return false; }
+
+        const auto concat_1_node = std::dynamic_pointer_cast<opset8::Concat>(reshape_1_node->input_value(1).get_node_shared_ptr());
+        if (!concat_1_node) return false;
+
+        const auto input_shape = reshape_1_node->get_input_shape(0);
+        if (!check_concat_1(concat_1_node, input_shape)) return false;
+
+        const auto concat_2_node = std::dynamic_pointer_cast<opset8::Concat>(pattern_to_output.at(concat_2).get_node_shared_ptr());
+        if (!concat_2_node) return false;
+
+        const auto new_spatial_shape = get_new_spatial_shape_from_concat_2(concat_2_node, input_shape);
+        if (new_spatial_shape.empty()) return false;
+
+        const auto ss_before_concat_1 = std::dynamic_pointer_cast<opset8::StridedSlice>(concat_1_node->input_value(0).get_node_shared_ptr());
+        const auto ss_before_concat_2 = std::dynamic_pointer_cast<opset8::StridedSlice>(concat_2_node->input_value(0).get_node_shared_ptr());
+        if (!ss_before_concat_1 || !ss_before_concat_2 || ss_before_concat_1.get() != ss_before_concat_2.get()) return false;
+
+        const auto shapeof_node = std::dynamic_pointer_cast<opset8::ShapeOf>(ss_before_concat_1->input_value(0).get_node_shared_ptr());
+        if (!shapeof_node) return false;
+
+        const auto node_before_shapeof = shapeof_node->input_value(0).get_node_shared_ptr();
```

Comment:
Please work with Output<Node> but not with shared_ptr<Node> because the type of node is unknown and if it has multi outputs you will get an exception during Interpolate node creation.

Reply:
you still can check nodes using outputs by calling method `get_node()` which returns raw ptr.

Reply:
Fixed.